packages/wallet-connectors/src/WalletConnect.ts

```diff
@@ -71,77 +73,109 @@ function accountTransactionPayloadToJson(data: AccountTransactionPayload) {
     });
 }
 
+function serializeInitContractParam(
+    initName: string,
+    parameters: SmartContractParameters | undefined,
+    schema: Schema | undefined
```

Comment:
Since these two either must both be defined or none, a more precise thing would be reflect that in the type and not allow a pair of (1 + X) types, instead just have 1 + (X, X). Then the checks are not needed.

Reply:
I agree; looking into it.

Reply:
done